sleap/nn/tracking.py

```diff
@@ -334,9 +454,40 @@ def get_candidates(
         return candidate_instances
 
 
+@attr.s(auto_attribs=True)
+class SimpleMaxTracksCandidateMaker:
+    """Class to generate instances based on the maximum number of tracks from prior frames."""
+
+    max_tracks: int = None
+    min_points: int = 0
+
+    @property
+    def uses_image(self):
+        return False
+
+    def get_candidates(
+        self,
+        track_matching_queue_dict: dict(),
+        *args,
+        **kwargs,
+    ) -> List[InstanceType]:
+        # Create set of matchable candidate instances from each track for max number of tracks.
+        candidate_instances = []
+        number_of_tracks = 0
+        for track, matched_instances in track_matching_queue_dict.items():
+            if track and number_of_tracks <= self.max_tracks:
```

Comment:
We shouldn't need to check for the `track` at this point since a `track is None` should not get added to the `track_matching_queue_dict`